### DISCUSSION TITLE
Possible to enforce the type of a template @Input() parameter?

### DISCUSSION THREAD

**Discussion by ThorstenKunz** [angular-eslint/angular-eslint#1639] [Q&A] [answered]
Is it possible somehow enforce that a parameter to an `@Input` binding in of template is of type TemplateRef and nothing else?
The component is a blackbox but it defines `@Input xzy: string | TemplateRef;`. Now if the parameter is a string the component is using some "innerHTML" assignments - which we really want to avoid so I only want to allow that it binds to TemplateRef types in our app.

Thanks

**Comment by JamesHenry** [comment, marked as answer]
Yes, you would write a custom rule which leverages the TypeScript compiler's type checking APIs:

https://typescript-eslint.io/developers/custom-rules#typed-rules

You can also leverage some of the existing utilities from `@angular-eslint/utils` in your rule implementation
